File: 算法训练营/07-递归/070/迭代法.py
#
# @lc app=leetcode.cn id=70 lang=python3
#
# [70] 爬楼梯
#

# @lc code=start


class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        r1, r2, r3 = 1, 2, 0
        for _ in range(3, n + 1):
            r3 = r1 + r2
            r1 = r2
            r2 = r3
        return r3

# @lc code=end

File: 算法训练营/20-字符串算法/387/迭代法.go
/*
 * @lc app=leetcode.cn id=387 lang=golang
 *
 * [387] 字符串中的第一个唯一字符
 */

// @lc code=start
func firstUniqChar(s string) int {
	var arr [256]rune
	for _, i := range s {
		arr[i]++
	}

	for no, i := range s {
		if arr[i] == 1 {
			return no
		}
	}
	return -1
}
// @lc code=end